move/squid/tests/squid/deepbook_v3_tests.move

```diff
@@ -0,0 +1,133 @@
+#[test_only]
+module squid::deepbook_v3_tests;
+
+use deepbook::pool::Pool;
+use deepbook::pool_tests;
+use squid::deepbook_v3;
+use squid::swap_info;
+use std::type_name;
+use sui::clock::Clock;
+use sui::coin::mint_for_testing;
+use sui::object::id_to_address;
+use sui::sui::SUI;
+use sui::test_scenario::{begin, end};
+use sui::test_utils::{destroy, assert_eq};
+use token::deep::DEEP;
+
+public struct USDC has store {}
+
+const OWNER: address = @0x1;
+
+#[test]
+fun test_serialize() {
+    let mut test = begin(OWNER);
+    let pool_id = pool_tests::setup_everything<SUI, USDC, SUI, DEEP>(&mut test);
+
+    test.next_tx(OWNER);
+    let swap_data = deepbook_v3::new_swap_data(
+        1,
+        id_to_address(&pool_id),
+        true,
+        100,
+        type_name::get<SUI>().into_string(),
+        type_name::get<USDC>().into_string(),
+        1,
+        true,
+    );
+    let swap_data_vec = std::bcs::to_bytes(&swap_data);
+    let data = std::bcs::to_bytes(&vector[swap_data_vec]);
+    let mut swap_info = swap_info::new(data, test.ctx());
+    let data2 = swap_info.get_data_estimating();
+    let swap_data2 = deepbook_v3::peel_swap_data(data2);
+    let swap_data = deepbook_v3::new_swap_data(
```

Comment:
Why reinstantiate this? The previous value is not used anywhere.

Reply:
good catch, fixed